lib/resources/submissions.js

```diff
@@ -65,12 +65,14 @@ const findMultipart = (files) => {
 };
 const forceAuthFailed = Problem.translate(Problem.user.insufficientRights, Problem.user.authenticationFailed);
 
-module.exports = (service, endpoint) => {
+module.exports = (service, endpoint, anonymousEndpoint) => {
 
   ////////////////////////////////////////////////////////////////////////////////
   // SUBMISSIONS (OPENROSA)
 
   const openRosaSubmission = (path, draft, getForm) => {
+    const endpointByTestPath = path.startsWith('/test/:key') ? anonymousEndpoint : endpoint;
```

Comment:
Would this code be more intuitive if the `endpoint` implementation was passed in rather than being derived from `path`?

e.g.

```js
const openRosaSubmission = (endpoint, path, draft, getForm) => {
```

or


```js
const openRosaSubmission = (path, { isDraft, requiresAuth }, getForm) => {
```

Reply:
I like the idea of making this an explicit argument (either of the suggestions above). :+1: